src/components/PlanOptionForm.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from "react";
+
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+const PlanOptionForm = ({ show, plans, planName, onSubmit, onClose }) => {
+  const [submissionDisabled, setSubmissionDisabled] = useState(false);
+
+  const handleSubmit = () => {
+    onSubmit(document.querySelector('input[name="plan-name"]').value.trim());
```

Comment:
Maybe create a getter function for `document.querySelector('input[name="plan-name"]').value` since it's accessed a few time.